network/message/authorization.go

```diff
@@ -0,0 +1,364 @@
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideApprovalsByChunk: {flow.RoleVerification}, // channel alias ProvideApprovalsByChunk  = RequestApprovalsByChunk
+
+		},
+	}
+
+	// generic entity exchange engines
+	authorizationConfigs[EntityRequest] = MsgAuthConfig{
+		Name: EntityRequest,
+		Type: func() interface{} {
+			return new(messages.EntityRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.RequestReceiptsByBlockID: {flow.RoleConsensus},
+			channels.RequestCollections:       {flow.RoleAccess, flow.RoleExecution},
+		},
+	}
+	authorizationConfigs[EntityResponse] = MsgAuthConfig{
+		Name: EntityResponse,
+		Type: func() interface{} {
+			return new(messages.EntityResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideApprovalsByChunk:  {flow.RoleVerification},
+			channels.ProvideReceiptsByBlockID: {flow.RoleExecution},
+		},
+	}
+
+	// testing
+	authorizationConfigs[TestMessage] = MsgAuthConfig{
+		Name: TestMessage,
+		Type: func() interface{} {
+			return new(message.TestMessage)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.TestNetworkChannel: flow.Roles(),
+			channels.TestMetricsChannel: flow.Roles(),
+		},
+	}
+
+	// DKG
+	authorizationConfigs[DKGMessage] = MsgAuthConfig{
+		Name: DKGMessage,
+		Type: func() interface{} {
+			return new(messages.DKGMessage)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.DKGCommittee: {flow.RoleConsensus},
+		},
+	}
+}
+
+// GetMessageAuthConfig checks the underlying type and returns the correct
+// message auth Config.
+// Expected error returns during normal operations:
+//  * ErrUnknownMsgType : if underlying type of v does  not match any of the known message types
+func GetMessageAuthConfig(v interface{}) (MsgAuthConfig, error) {
+	switch v.(type) {
+	// consensus
+	case *messages.BlockProposal:
+		return authorizationConfigs[BlockProposal], nil
+	case *messages.BlockVote:
+		return authorizationConfigs[BlockVote], nil
+
+	// protocol state sync
+	case *messages.SyncRequest:
+		return authorizationConfigs[SyncRequest], nil
+	case *messages.SyncResponse:
+		return authorizationConfigs[SyncResponse], nil
+	case *messages.RangeRequest:
+		return authorizationConfigs[RangeRequest], nil
+	case *messages.BatchRequest:
+		return authorizationConfigs[BatchRequest], nil
+	case *messages.BlockResponse:
+		return authorizationConfigs[BlockResponse], nil
+
+	// cluster consensus
+	case *messages.ClusterBlockProposal:
+		return authorizationConfigs[ClusterBlockProposal], nil
+	case *messages.ClusterBlockVote:
+		return authorizationConfigs[ClusterBlockVote], nil
+	case *messages.ClusterBlockResponse:
+		return authorizationConfigs[ClusterBlockResponse], nil
+
+	// collections, guarantees & transactions
+	case *flow.CollectionGuarantee:
+		return authorizationConfigs[CollectionGuarantee], nil
+	case *flow.TransactionBody:
+		return authorizationConfigs[TransactionBody], nil
+	case *flow.Transaction:
```

Comment:
## ❓ 

I don't think `authorizationConfigs[Transaction]` is specified, is it?

Reply:
You are right we don't ever transmit the full Transaction

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/18f74004d97bf1aceaf1e5931b3f41ba070898f6
https://github.com/onflow/flow-go/pull/2648/commits/017ee461ac1a7c938fdd16e4ece3566a522cac41